cmd/infra/powervs/validate.go

```diff
@@ -132,6 +132,31 @@ func validateVpc(vpcName string, resourceGroupID string, v1 *vpcv1.VpcV1) (vpc *
 	}
 
 	if vpc != nil {
+		vpcId := vpc.DefaultSecurityGroup.ID
+		var e error
+		vpcSg, _, e := v1.GetSecurityGroup(&vpcv1.GetSecurityGroupOptions{ID: vpcId})
+		if err != nil {
+			err = fmt.Errorf("error retrieving security group of vpc %w", e)
```

Comment:
why are we not returning from here if err?

Reply:
added return here.